pkg/cli/upgrade/steps.go

```diff
@@ -108,6 +115,20 @@ func (u *Upgrade) upgradeCustomResourceDefinitions(ctx context.Context) error {
 	return installer.Install(ctx)
 }
 
+// legacyUpgradeCRDs upgrades the CRDs for any version below 1.9.0.
+// This is kept for backward compatibility with versions below 1.9.0.
+func (u *Upgrade) legacyUpgradeCRDs(ctx context.Context) error {
+	manifests, err := u.helmInstaller.RenderTemplates(ctx)
+	if err != nil {
+		return fmt.Errorf("could not render Helm templates: %w", err)
+	}
+	crds, err := manifests.GetCRDs()
+	if err != nil {
+		return fmt.Errorf("could not get CRDs: %w", err)
+	}
+	return u.kubeConnector.ApplyManifestFile(ctx, helmutils.YAMLStringsToBytes(crds), common.SystemNamespace)
+}
+
```

Comment:
So this is the way we handled CRDs upgrade prior `1.9.0`. But since the `1.10.0` will allow to upgrade from `1.8.0` to `1.8.1` which is older than `1.9.0` we need to keep the method for backward compatibility.